lib/practice_exercises.rb

```diff
@@ -1,13 +1,37 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: guess--O(n). Even it's a really big array, even though it's divided in half, will still take awhile to process through.
+# Space Complexity: I'm not confident at all on this one. O(n)? I first put O(1), but I do think if it's a long array, then teh space need grows with n?
 def is_palindrome(string)
```

Comment:
👍 , however your space complexity is O(n) because `gsub` and `downcase` are making copies of the original string.